spec/enumerable_spec.rb

```diff
@@ -0,0 +1,113 @@
+require_relative '../enumerables'
+
+describe Enumerable do
+  let(:array) { [1, 2, 3] }
+  let(:hash) { { 'a' => 'enum', 'b' => 'module', 'c' => 'spec' } }
+
+  describe '#my_each' do
+    it 'return each value in an array' do
+      i = 0
+      array.my_each do |item|
+        expect(item).to eql(array[i])
+        i += 1
+      end
+    end
+
+    it 'return each value in a hash' do
+      hash.my_each do |key, value|
+        expect(value).to eql(hash[key])
+      end
+    end
+  end
+
+  describe '#my_each_with_index' do
+    it 'returns a value from an index in an array' do
+      i = 0
+      array.my_each_with_index do |value, index|
+        expect(index).to eql(i)
+        expect(value).to eql(array[i])
+        i += 1
+      end
+    end
+  end
```

Comment:
Specs should cover positive and negative scenarios. In both `my_each` and `my_each_with_index`, the positive scenarios are when a block is given. Cover the scenarios for when a block is not given for each of these methods. See [screenshot ](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from pathwright.